app/infrastructure/persistence/repositories/auditable_repo.py

```
@@ -147,3 +148,69 @@ async def emit_custom_audit(
+
+class TenantScopedRepository(AuditableRepository[ModelType]):
+    """Repository that enforces tenant isolation.
+
+    All reads (get_by_id, get_all) and writes (create, update, delete) are
+    scoped to a single tenant_id. Pass tenant_id at construction; the repo
+    then auto-adds tenant filtering and rejects cross-tenant writes.
+    """
+
+    def __init__(
+        self,
+        db: "AsyncSession",
```

Ruff (0.15.0)

[warning] Remove quotes from type annotation

Remove quotes

(UP037)